internal/provisioner/kops_utils.go

```diff
@@ -110,3 +111,52 @@ func getLoadBalancerEndpoint(ctx context.Context, namespace string, logger log.F
 		}
 	}
 }
+
+// GetNGINXLoadBalancerEndpoint returns the load balancer endpoint of the NGINX service.
+func (provisioner *KopsProvisioner) GetNGINXLoadBalancerEndpoint(cluster *model.Cluster, namespace string) (string, error) {
+	logger := provisioner.logger.WithFields(log.Fields{
+		"cluster":         cluster.ID,
+		"nginx-namespace": namespace,
+	})
+	kops, err := kops.New(provisioner.s3StateStore, logger)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create kops wrapper")
+	}
+	defer kops.Close()
+
+	kopsMetadata, err := model.NewKopsMetadata(cluster.ProvisionerMetadata)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse provisioner metadata")
+	}
+
+	if kopsMetadata.Name == "" {
+		logger.Infof("Cluster %s has no name, assuming cluster installation never existed.", cluster.ID)
+		return "", nil
+	}
+
+	err = kops.ExportKubecfg(kopsMetadata.Name)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to export kubecfg")
+	}
+
+	k8sClient, err := k8s.New(kops.GetKubeConfigPath(), logger)
+	if err != nil {
+		return "", err
+	}
+
+	services, err := k8sClient.Clientset.CoreV1().Services(namespace).List(metav1.ListOptions{})
```

Comment:
I'm not sure how big list can be but can't we take advantage `ListOptions` for filtering?

Reply:
@gsagula I don't think its needed because it filters initially by namespace and in the nginx case its gets only one service back.

Reply:
No worries then, thanks!